augur/utils.py

```diff
@@ -111,25 +112,23 @@ def is_date_ambiguous(date, ambiguous_by="any"):
         "X" in day and ambiguous_by in ("any", "day")
     ))
 
-def get_numerical_date_from_value(value, fmt=None, min_max_year=None, raise_error=True):
-    if type(value)!=str:
-        if raise_error:
-            raise ValueError(value)
-        else:
-            numerical_date = None
-    elif 'XX' in value:
+def get_numerical_date_from_value(value, fmt=None, min_max_year=None):
+    value = str(value)
+    if re.match(r'^-*\d+\.\d+$', value):
+        # numeric date which can be negative
+        return float(value)
+    if value.isnumeric():
+        # year-only date is ambiguous
+        value = fmt.replace('%Y', value).replace('%m', 'XX').replace('%d', 'XX')
```

Comment:
Yeah, I also couldn't figure out the motivation for the `raise_error` parameter – supposedly it raises an error that is then handled in `get_numerical_dates` here:

https://github.com/nextstrain/augur/blob/a85194c243db8d85e6fc06ea2d614e0b6095a0c4/augur/utils.py#L147-L152

I know `get_numerical_dates` is used outside of `augur filter` – do you see any concern of removing this?

If not, I should also remove the try/except but I'll wait for your input first.

Reply:
The logic to check the value type was originally added as part of a "[small fixes](https://github.com/nextstrain/augur/commit/67854ec0712ddc4f4e3557d01309b9848e2eba62)" commit. When I refactored the adjacent logic into its own function, [I changed the warning to an exception](https://github.com/nextstrain/augur/commit/43e6ded49b745e158d74b14466711ae38673ad4d#diff-8f66173c8ddc9d2afae3a0146d3abfcf6004537354e5ba7cec8fdf5186c3c0f0R99-R100), to avoid silent failures (assuming people would not see the warning in the logs, but would notice when their script crashes).

That said, none of [the code I can find in our organization that calls `get_numerical_dates`](https://github.com/search?q=org%3Anextstrain+get_numerical_dates&type=code) expects or handles a `ValueError`. The rest of the function tolerates invalid or missing information, returning `None` when no valid date can be parsed. Although this implementation obscures potentially problematic metadata, your implementation in this PR is most consistent with the rest of the function and I don't think anyone will miss the `ValueError` exception.

Reply:
Sounds good, I've removed the part that handled this exception: https://github.com/nextstrain/augur/pull/841/commits/bfb21dae711dd6267673bcd9e8f7a284bd84a909